linera-core/src/client/mod.rs

```diff
@@ -4129,11 +4133,14 @@ impl<Env: Environment> ChainClient<Env> {
         Ok(())
     }
 
-    /// Returns whether the given ownership includes anyone whose secret key we don't have.
-    fn has_other_owners(&self, ownership: &ChainOwnership) -> bool {
-        ownership
-            .all_owners()
-            .any(|owner| Some(owner) != self.preferred_owner.as_ref())
+    /// Returns whether the given ownership includes anyone whose secret key we don't have, or
+    /// any regular (not super) owners. Returns `false` only if we are the only owner and a super.
+    fn has_other_or_regular_owners(&self, ownership: &ChainOwnership) -> bool {
+        !ownership.owners.is_empty()
+            || ownership
+                .super_owners
+                .iter()
+                .any(|owner| Some(owner) != self.preferred_owner.as_ref())
```

Comment:
Maybe this method would be simpler if it was on `ChainOwnership` and `self.preferred_owner` was the argument? Something like `is_singular_and_super_owner(self.preferred_owner)`? Right now it reads kinda ... backwards.